modules/core/message_functions.php

```diff
@@ -449,7 +449,7 @@ class HTMLToText {
 
     function __construct($html) {
         $doc = new DOMDocument();
-        $doc->loadHTML(html_entity_decode($html, ENT_QUOTES | ENT_HTML5, 'UTF-8'));
+        $doc->loadHTML(mb_convert_encoding($html, 'HTML-ENTITIES', 'UTF-8'));
```

Comment:
use https://www.php.net/manual/en/function.htmlentities.php instead is the goal is to prevent HTML injection by converting special characters to HTML entities

Reply:
Great, thank you for the guidance.